include/openssl/e_os2.h

```diff
@@ -350,6 +350,14 @@ typedef unsigned __int64 uint64_t;
 #  endif
 # endif
 
+# if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
+#  define ossl_noreturn _Noreturn
+# elif defined(__GNUC__)
```

Comment:
Do you mean the C11-compliant `_Noreturn` above or the `__attribute__((noreturn))` below? The former [appeared in GCC 4.7](https://gcc.gnu.org/gcc-4.7/changes.html), and I'm not quite sure about the latter.

I guess the best option would be to order the directives so that we first test for GCC < 4.7 and then for C11.